cliquet/__init__.py

```diff
@@ -124,23 +125,34 @@ def load_default_settings(config):
 def handle_api_redirection(config):
     """Add a view which redirects to the current version of the API.
     """
-    # Disable the route prefix passed by the app.
-    route_prefix = config.route_prefix
-    config.route_prefix = None
-
-    def _redirect_to_version_view(request):
-        raise HTTPTemporaryRedirect(
-            '/%s/%s' % (route_prefix, request.matchdict['path']))
+    settings = config.get_settings()
+    redirect_enabled = settings['cliquet.version_prefix_redirect_enabled']
+    version_prefix_redirection_enabled = asbool(redirect_enabled)
 
     # Redirect to the current version of the API if the prefix isn't used.
-    config.add_route(name='redirect_to_version',
-                     pattern='/{path:(?!%s).*}' % route_prefix)
+    # Do not redirect if cliquet.version_prefix_redirect_enabled is set to
+    # False.
+    if version_prefix_redirection_enabled:
+
+        def _redirect_to_version_view(request):
+            raise HTTPTemporaryRedirect(
+                '/%s/%s' % (route_prefix, request.matchdict['path']))
+
+        # Disable the route prefix passed by the app.
+        route_prefix = config.route_prefix
+        config.route_prefix = None
+
+        config.add_route(name='redirect_to_version',
+                         pattern='/{path:(?!%s).*}' % route_prefix)
+
+        config.add_view(view=_redirect_to_version_view,
+                        route_name='redirect_to_version',
+                        permission=NO_PERMISSION_REQUIRED)
 
-    config.add_view(view=_redirect_to_version_view,
-                    route_name='redirect_to_version',
-                    permission=NO_PERMISSION_REQUIRED)
+        config.route_prefix = route_prefix
 
-    config.route_prefix = route_prefix
+    else:
+        return
```

Comment:
This statement does not bring anything. What I had in mind when I mentionned to invert the condition was this:

Replace : 

```python
if condition:
    # very
    # long
    # piece
    # of
    # indented
    # code
```

by : 

```python
if not condiition:
    return

# very
# long
# piece
# of
# code
```